src/public/modules/forms/helpers/ndjsonStream.ts

```diff
@@ -0,0 +1,68 @@
+// Modified fork of https://github.com/canjs/can-ndjson-stream to enqueue
+// the string immediately without a JSON.parse() step, as the stream payload
+// is to be decrypted by the decryption worker.
+
+// Note that this code assumes a polyfill of TextDecoder is available to run in IE11.
+
+export const ndjsonStream = (
+  response: ReadableStream<Uint8Array>,
+): ReadableStream => {
+  // For cancellation
+  let maybeReader: ReadableStreamDefaultReader<Uint8Array> | undefined
+  let shouldCancel = false
+  return new ReadableStream({
+    start: function (controller) {
+      const reader = response.getReader()
+      maybeReader = reader
+      const decoder = new TextDecoder()
+      let data_buf = ''
+
+      return reader
+        .read()
+        .then(function processResult(result): Promise<void> | undefined | void {
+          if (shouldCancel) {
+            return
+          }
+
+          if (result.done) {
+            data_buf = data_buf.trim()
+            if (data_buf.length !== 0) {
+              try {
+                controller.enqueue(data_buf)
+              } catch (e) {
+                controller.error(e)
+                return
+              }
+            }
+            return controller.close()
+          }
+
+          // Read the input in as a stream and split by newline and trim
+          const data = decoder.decode(result.value, { stream: true })
+          data_buf += data
+          const lines = data_buf.split('\n').map((line) => line.trim())
+
+          // Only append if there is content available
+          lines.forEach((line) => {
+            if (line) {
+              try {
+                controller.enqueue(line)
+              } catch (e) {
+                controller.error(e)
+                shouldCancel = true
+                return reader.cancel()
+              }
+            }
+          })
+
+          data_buf = lines[lines.length - 1]
+          return reader.read().then(processResult)
+        })
+    },
+    cancel: function (reason) {
+      console.log('Cancel registered due to ', reason)
+      shouldCancel = true
+      maybeReader && maybeReader.cancel()
```

Comment:
consider sticking to `if (maybeReader) maybeRead.cancel()` for style consistency

Reply:
ok sure, this works too!